packages/mui-material/src/Avatar/Avatar.js

```diff
@@ -183,7 +183,9 @@ const Avatar = React.forwardRef(function Avatar(inProps, ref) {
         {...imgProps}
       />
     );
-  } else if (childrenProp != null && childrenProp !== '' && typeof childrenProp !== 'boolean') {
+
+    // Consider the children valid to render as long as not flasy and not 0.
+  } else if (!(!childrenProp && childrenProp !== 0)) {
```

Comment:
In this case we are ignoring `-0`. Is it ok?

Reply:
`0 === -0`